pkg/model/keycloak_route.go

```diff
@@ -53,6 +53,13 @@ func KeycloakRouteReconciled(cr *kc.Keycloak, currentState *v1.Route) *v1.Route
 	return reconciled
 }
 
+func getTLSTerminationType(cr *kc.Keycloak) v1.TLSTerminationType {
+	if cr.Spec.ExternalAccess.TLSTermination == kc.ReencryptTLSTerminationType {
```

Comment:
I'm not much of a go coder, but this doesn't look right ;)

Reply:
No, that's fine :) The `ReencryptTLSTerminationType` is a type of a String.

Reply:
Yes, but it says if the configured option is equal to reencrypt, then return passthrough?

Reply:
ooh yes! You're absolutely right... last minute refactoring... arghhhhhh

Reply:
Fixed.

Reply:
@slaskawi I think this should be actually the other way around. This way it defaults to `passthrough` and we need `reencrypt`. 

```suggestion
	if cr.Spec.ExternalAccess.TLSTermination == kc.PassthroughTLSTerminationType {
	    return "passtrhough"
	}
	return "reencrypt"
```